api/enki.py

```diff
@@ -233,126 +222,83 @@ def recent_activity(self, minutes=5):
+        """
+        url = self.base_url + self.item_endpoint
+        args = dict(
+            method="getItem",
+            recordid=enki_id,
+            lib=self.library_id,
+            size="large",
+        )
+        response = self.request(url, params=args)
+        try:
+            data = json.loads(response.content)
+        except ValueError, e:
+            # This is most likely a 'not found' error.
+            return None
+
+        book = data.get('result', {})
+        if book:
+            return BibliographicParser().extract_bibliographic(book)
+        return None
+
+    def get_all_titles(self, strt=0, qty=10):
+        """Retrieve a single page of items from the Enki collection.
 
-    def availability(self, patron_id=None, since=None, title_ids=[], strt=0, qty=2000):
-        """Get *all* titles in a paginated list.
+        Iterating over the entire collection is very expensive and
+        should only happen during initial data population.
 
-        This is a very expensive API call and should only be used
-        during initial data population.
+        :yield: A sequence of Metadata objects, each with a
+            CirculationData attached.
         """
-        qty = 10
         self.log.debug ("requesting : "+ str(qty) + " books starting at econtentRecord" +  str(strt))
-        url = str(self.base_url) + str(self.availability_endpoint)
+        url = str(self.base_url) + str(self.list_endpoint)
         args = dict()
         args['method'] = "getAllTitles"
         args['id'] = "secontent"
         args['strt'] = strt
         args['qty'] = qty
         args['lib'] = self.library_id
         response = self.request(url, params=args)
-        return response
-
-    @classmethod
-    def create_identifier_strings(cls, identifiers):
-        identifier_strings = []
-        for i in identifiers:
-            if isinstance(i, Identifier):
-                value = i.identifier
-            else:
-                value = i
-            identifier_strings.append(value)
-
-        return identifier_strings
+        for metadata in BibliographicParser().process_all(response.content):
+            yield metadata
 
     @classmethod
-    def parse_token(cls, token):
-        data = json.loads(token)
-        return data['access_token']
-
-    def _make_request(self, url, method, headers, data=None, params=None,
-                      **kwargs):
-        """Actually make an HTTP request."""
-        try:
-            return HTTP.request_with_timeout(
-                method, url, headers=headers, data=data,
-                params=params, timeout=90, **kwargs
-            )
-        except RequestTimedOut, e:
-            self.log.info("Request to %s timed out once. Trying a second time.", url)
-            return HTTP.request_with_timeout(
-                method, url, headers=headers, data=data,
-                params=params, timeout=90, **kwargs
-            )
-
-    def reaper_request(self, identifier):
```

Comment:
This is just a getItem request; the API class doesn't need to know what will be done with the results.